src/main/java/org/valdroz/vscript/BaseNode.java

```diff
@@ -330,7 +330,17 @@ public Variant execute(VariantContainer variantContainer) {
                 result = Variant.fromArray(arrItems);
             }
             break;
-
+            case NT_MF_IF: {
+                List<Variant> parameterItems = Lists.newArrayList();
+                if (params != null) {
+                    params.forEach(itemNode -> parameterItems.add(itemNode.execute(variantContainer)));
+                }
+                if (parameterItems.size() != 3) {
```

Comment:
Validation should be done before any parameter nodes are evaluated

Reply:
updated based on your comments. thanks